tensorflow/python/keras/layers/normalization.py

```diff
@@ -997,29 +1017,67 @@ def call(self, inputs):
     input_shape = inputs.shape
     ndims = len(input_shape)
 
-    # Calculate the moments on the last axis (layer activations).
-    mean, variance = nn.moments(inputs, self.axis, keep_dims=True)
-
-    # Broadcasting only necessary for norm where the axis is not just
-    # the last dimension
-    broadcast_shape = [1] * ndims
-    for dim in self.axis:
-      broadcast_shape[dim] = input_shape.dims[dim].value
-    def _broadcast(v):
-      if (v is not None and len(v.shape) != ndims and
-          self.axis != [ndims - 1]):
-        return array_ops.reshape(v, broadcast_shape)
-      return v
-    scale, offset = _broadcast(self.gamma), _broadcast(self.beta)
+    if not self.fused:
+      # Calculate the moments on the last axis (layer activations).
+      mean, variance = nn.moments(inputs, self.axis, keep_dims=True)
+
+      # Broadcasting only necessary for norm where the axis is not just
+      # the last dimension
+      broadcast_shape = [1] * ndims
+      for dim in self.axis:
+        broadcast_shape[dim] = input_shape.dims[dim].value
+      def _broadcast(v):
+        if (v is not None and len(v.shape) != ndims and
+            self.axis != [ndims - 1]):
+          return array_ops.reshape(v, broadcast_shape)
+        return v
+      scale, offset = _broadcast(self.gamma), _broadcast(self.beta)
+
+      # Compute layer normalization using the batch_normalization function.
+      outputs = nn.batch_normalization(
+          inputs,
+          mean,
+          variance,
+          offset=offset,
+          scale=scale,
+          variance_epsilon=self.epsilon)
+    else:
+      # Squeeze self.axis to C in NCHW of squeezed_shape 
+      squeezed_shape = [1] * 4
+      axis = sorted(self.axis)
+      for dim in range(0, ndims):
```

Comment:
Can u add some comment here to explain the details? If I understand it correctly, any dim before axis is compressed to N, any dim within axis is compressed to C, and dim after axis is compressed H, and W is not used?